app/models/person.rb

```diff
@@ -50,17 +53,101 @@ class Person < ActiveRecord::Base
   has_many :secondary_agent_loans, class_name: 'Loan', foreign_key: :secondary_agent_id
   has_many :representative_loans,  class_name: 'Loan', foreign_key: :representative_id
 
-  has_one :user, foreign_key: :profile_id
+  has_one :user, foreign_key: :profile_id, autosave: true
 
   validates :division_id, presence: true
   validates :first_name, presence: true
 
+  validate :division_role_valid
+  validate :user_valid
+
+  # Transient attributes to fascilitate user managment
+  attr_writer :owning_division_role
+  attr_accessor :password, :password_confirmation
+
+  before_save :update_full_name
+  before_save :update_user
+  after_save :handle_roles
+  after_save :clean_up_passwords
+
+  def update_full_name
+    self.name = "#{first_name} #{last_name}"
+  end
+
+  # Lazy evaluation getter
+  def owning_division_role
+    @owning_division_role = resolve_owning_division_role unless defined? @owning_division_role
+    @owning_division_role
+  end
+
+  def owning_division_role_label
+    label_key = owning_division_role || 'none'
+    I18n.t("people.roles.#{label_key}")
+  end
+
+  def resolve_owning_division_role
+    if user
+      role = user.roles.find_by(resource_type: 'Division', resource_id: division_id)
+      return role.name.to_sym if role
+    end
+    nil
+  end
+
+  def user_required?
+    has_system_access
```

Comment:
Can/should add `?` at end of this line by convention. Rails supports automatically.